### ISSUE TITLE
DietPi-Software | myMPD, fork of YMPD with additional features

### ISSUE THREAD

**Issue by jcorporation** [MichaIng/DietPi#2032]
#### Is your feature request related to a problem? Please describe:
- I am programming a fork of ympd named myMPD (on github). myMPD supports play statistics and song voting and uses the mpd sticker functionality for this features. The native mpd from the repository support stickers, but the dietpi optimized mpd not.

#### Describe the solution you'd like:
- Compile dietpi custom mpd with sticker support

#### Describe alternatives you've considered:
- Implementing a own database is the only alternative, but myMPD should use only mpd as source.

#### Additional context:
- none

**Comment by Fourdee** [comment]
@jcorporation 

Hi,

Many thanks for the request 👍 

YMPD is a favorite of mine, so i'll definitely be adding this to ```dietpi-software```.

REF: https://github.com/jcorporation/myMPD#mympd

---

Compiles, MPD + sticker support (```mpd-0.20.21.tar.xz```):
- 🈯️ ARMv6 | Many hours later
- 🈯️ ARMv7
- 🈯️ ARMv8
- 🈯️ x86_64

---

myMPD:
- Sourcebuild
- 148
- ```/etc/mpd.conf```
```
# The location of the sticker database.  This is a database which
# manages dynamic information attached to songs.
#
sticker_file                   "/var/lib/mpd/sticker.sql"
```

Run and cfg
```
mympd /etc/mympd/mympd.conf
```

**Comment by jcorporation** [comment]
Many thanks for the fast response. I am looking forward to the 6.15 release. It would be great if you package myMPD for dietpi. If you need help don't hesitate to ask.

**Comment by Fourdee** [comment]
```folder.jpg``` Testing 🈯️ 
![image](https://user-images.githubusercontent.com/7077026/44554388-aabf6300-a728-11e8-9520-e520983fa4e0.png)

**Comment by Fourdee** [comment]
- 🈯️ Online doc: https://dietpi.com/phpbb/viewtopic.php?f=8&t=5&p=13978#p13978
- 🈯️ https://dietpi.com/software addition

May need to disable SSL and its redirect:
```
Aug 24 00:44:33 DietPi mympd[6863]: myMPD started on http port 1333
Aug 24 00:44:33 DietPi mympd[6863]: myMPD started on ssl port 443
Aug 24 00:44:33 DietPi mympd[6863]: MPD Connecting to 127.0.0.1: 6600
Aug 24 00:44:33 DietPi mympd[6863]: MPD connected.
Aug 24 00:45:37 DietPi mympd[6863]: Redirecting to https://DietPi:443/
```
- redirects to https://hostname (could not be unique)
- uses up the https default site? (could use ```sslport = 1334```, but the above is more an issue for multiple installs with same hostname)
- 🈯️ We'll disable SSL + redirect for now. Referenced config and its options to enable SSL in the online doc above.

**Comment by Fourdee** [comment]
@jcorporation 

>  It would be great if you package myMPD for dietpi

We've gone for a sourcebuild on this one, using the master branch of your project. For now. If its popular, we can do a install binary for each CPU arch.

---

Installation nearly completed. I just need to run some test installs on various archs.

myMPD. Really impressive on what i've seen so far. Its YMPD and then some! Great work 👍

**Comment by Fourdee** [comment]
@jcorporation 

Once minor issue I have noticed:
When clicking the 'Update Database' option, the drop down menu does not disappear and feels it does nothing. However, the database appears to be functional once done?
NB: no ```folder.jpg``` for this item.
![image](https://user-images.githubusercontent.com/7077026/44558790-14963780-a73e-11e8-9902-efd6f0ab0aae.png)

**Comment by Fourdee** [comment]
Completed. Available in DietPi v6.15 update.

v6.15 should be 1-2 weeks away.

**Comment by jcorporation** [comment]
@Fourdee:

I currently merged the devel branch into master for the 4.0.0 release of myMPD.
This fixes hopefully the issue with "no update database progress in ui" and the issue with the ssl redirect (now myMPD uses the original host header for redirection).

I hope this is the right place for this message to you.

Many thanks in advance.

**Comment by Fourdee** [comment]
@jcorporation 

> I currently merged the devel branch into master for the 4.0.0 release of myMPD.
This fixes hopefully the issue with "no update database progress in ui" and the issue with the ssl redirect (now myMPD uses the original host header for redirection).

Excellent, many thanks 👍 

As v6.15 is still in development, once released, the user will install myMPD from master branch. (hopefully release v6.15 end of this week 😃 )
